Cython/Build/IpythonMagic.py

```diff
@@ -109,14 +109,17 @@ def encode_fs(name):
 
 def prepare_captured(captured):
     captured_bytes = captured and captured().strip()
-    if captured_bytes:
-        for enc in [sys.getdefaultencoding(), 'latin-1']:
-            try:
-                return captured_bytes.decode(enc)
-            except UnicodeDecodeError:
-                pass
-    # returns as bytes, as it couldn't be decoded
-    return captured_bytes
+    if not captured_bytes:
+        return None
+    tried = {None}
+    for encoding in [sys.getdefaultencoding(), sys.stdout.encoding, sys.stdin.encoding]:
```

Comment:
With `sys.stdout` being replaced/redirected, `sys.stdout` could have no `encoding`-attribute. However, in Jupyter `sys.stdout` has `encoding` atribute - so the problem is probably due to using ` IPython.utils.io.capture_output`. Maybe using `sys.__stdout__` is more robust?